src/app/window-utils.ts

```diff
@@ -1077,6 +1088,35 @@ export const updateFeaturesForCloudConfig = async (
       }
     }
   }
+
+  // SDA auto updater
+  logger.info(`window-utils: initiate auto update?`, isAutoUpdateEnabled);
+  if (isAutoUpdateEnabled) {
+    if (!autoUpdateIntervalId) {
+      // randomised to avoid having all users getting SDA update at the same time
+      autoUpdateIntervalId = setInterval(async () => {
+        const { lastAutoUpdateCheckDate } = config.getUserConfigFields([
+          'lastAutoUpdateCheckDate',
+        ]);
+        logger.info(
+          `window-utils: is last check date > auto update check interval?`,
+          lastAutoUpdateCheckDate,
+          autoUpdateCheckInterval,
+        );
+        // Compare the current date and user config last auto update checked date
+        // and if it is greater that autoUpdateCheckInterval we check for new updates
+        if (
+          getDifferenceInDays(new Date(), new Date(lastAutoUpdateCheckDate)) >
```

Comment:
`lastAutoUpdateCheckDate ` not being set the first time,  `new Date(lastAutoUpdateCheckDate)`. will fail.
I think it's better to check if we have a valid value when we get it from the config file. If it's not a valid one (empty string, undefined) then we can set the value in config file and run `autoUpdate.checkUpdates()` , wdyt?